tests/compiler/std/typedarray.ts

```diff
@@ -693,3 +693,47 @@ testArrayWrap<Int64Array, i64>();
 testArrayWrap<Uint64Array, u64>();
 testArrayWrap<Float32Array, f32>();
 testArrayWrap<Float64Array, f64>();
+
+function valuesEqual<T extends ArrayBufferView>(target: T, compare: valueof<T>[]): void {
+  let len = target.length;
+  assert(len == compare.length);
+  for (let i = 0; i < len; i++) {
+    let vala = target[i];
+    let valb = unchecked(compare[i]);
+    if (vala != valb) {
+      trace(nameof<T>(), 3, i, <f64>vala, <f64>valb);
+      assert(false);
+    }
+  }
+}
+
+let setSource1: i32[] = [1, 2, 3];
+let setSource2: f32[] = [4, 5, 6];
+let setSource3: f64[] = [Infinity, NaN, -Infinity];
```

Comment:
I guess need add `i64[]` and typed array sources as well

Reply:
Okay. I'll add a few more.

Reply:
Came here looking for an i16->i8 test. Might make sense to have a signed and unsigned small int here, a signed and unsigned normal i32 and a signed and unsigned i64.

Reply:
Please forgive me for finally getting around to this. I added a few more tests and actually found an edge case involving `Uint8ClampedArray`.

I would be happy to add some more tests if you'd like! I just need to get around to them tomorrow.

Can I get some feedback on the ones I put in?

Reply:
Looks good now with the additional Uint8Array and Int16Array cases.